src/analytics/formTracker/formTracker.ts

```diff
@@ -1,6 +1,182 @@
-function formTracker() {
-  console.log("running formTracker");
-  return "";
-}
+import { validateParameter } from "../../utils/validateParameter";
+import { BaseTracker } from "../baseTracker/baseTracker";
+import { FormEventInterface, FormField } from "./formTracker.interface";
+
+export class FormResponseTracker extends BaseTracker {
+  eventName: string = "form_response";
+  eventType: string = "event_data";
+
+  constructor() {
+    super();
+    this.initialiseEventListener();
+  }
+
+  /**
+   * Initialise the event listener for the document.
+   *
+   */
+  initialiseEventListener() {
+    document.addEventListener(
+      "submit",
+      this.trackFormResponse.bind(this),
+      false,
+    );
+  }
+
+  trackFormResponse(event: SubmitEvent): boolean {
+    const form = document.forms[0];
+    let fields: FormField[] = [];
+
+    if (form && form.elements) {
+      fields = this.getFields(form);
+    } else {
+      return false;
+    }
+
+    if (!fields.length) {
+      console.log("form or fields not found");
+      return false;
+    }
+
+    const formResponseTrackerEvent: FormEventInterface = {
+      event: this.eventType,
+      event_data: {
+        event_name: this.eventName,
+        type: validateParameter(this.getFieldType(fields), 100),
+        url: "undefined",
+        text: validateParameter(this.getFieldValue(fields), 100),
+        section: validateParameter(this.getFieldLabel(), 100),
+        action: validateParameter(this.getButtonLabel(event), 100),
+        external: "undefined",
+      },
+    };
 
-export default formTracker;
+    try {
+      this.pushToDataLayer(formResponseTrackerEvent);
+      return true;
+    } catch (err) {
+      console.error("Error in trackFormResponses", err);
+      return false;
+    }
+  }
+
+  /**
+   * Retrieves the selected fields from an HTML form.
+   *
+   * @param {HTMLFormElement} form - The HTML form element.
+   * @return {FormField[]} An array of selected form fields.
+   */
+  getFields(form: HTMLFormElement): FormField[] {
+    const selectedFields: FormField[] = [];
+    for (let i = 0; i < form.elements.length; i++) {
+      const element: HTMLInputElement = form.elements[i] as HTMLInputElement;
+
+      if (
+        element.type === "hidden" ||
+        element.type === "fieldset" ||
+        element.type === "submit"
+      ) {
+        continue;
+      }
+
+      if (
+        (element.type === "radio" || element.type === "checkbox") &&
+        element.checked
+      ) {
+        selectedFields.push({
+          id: element.id,
+          name: element.name,
+          value: document.querySelector(`label[for="${element.id}"]`)
+            ?.textContent
+            ? document
+                .querySelector(`label[for="${element.id}"]`)
+                ?.textContent?.trim()
+            : "undefined",
+          type: element.type,
+        });
+      } else if (element.type === "textarea" || element.type === "text") {
```

Comment:
we don't have a text input form in our demo app currently , do we need one for testing or is ok just to test with repos from pods